File: Composition/CS/Composition/Algorithm/Inheritance/LowPassAveragingAggregator.cs
using System.Collections.Generic;
using System.Linq;

namespace Algorithm.Inheritance
{
    public class LowPassAveragingAggregator : AveragingAggregator
    {
        public LowPassAveragingAggregator(IEnumerable<Measurement> measurements) : base(measurements)
        {
        }

        protected override IEnumerable<Measurement> FilterMeasurements(IEnumerable<Measurement> measurements)
        {
            return measurements.Where(m => m.X < 100 && m.Y < 100);
        }
    }
}

File: Composition/Java/Composition/src/main/java/algorithm/inhereitance/LowPassAveragingAggregator.java
package algorithm.inhereitance;

import java.util.Collection;

import algorithm.Measurement;
import algorithm.MeasurementUtil;

public class LowPassAveragingAggregator extends AveragingAggregator {

	public LowPassAveragingAggregator(Collection<Measurement> measurements) {
		super(measurements);
	}

	@Override
	protected Collection<Measurement> filterMeasurements() {
		return MeasurementUtil.WhereLessThanXandY(measurements, 100, 100);
	}
}
